Sources/TuistSupport/SwiftPackageManager/SwiftPackageManagerController.swift

```diff
@@ -53,16 +55,22 @@ public final class SwiftPackageManagerController: SwiftPackageManagerControlling
         self.fileHandler = fileHandler
     }
 
-    public func resolve(at path: AbsolutePath, printOutput: Bool) throws {
-        let command = buildSwiftPackageCommand(packagePath: path, extraArguments: ["resolve"])
+    public func resolve(at path: AbsolutePath, arguments: [String], printOutput: Bool) throws {
+        let command = buildSwiftPackageCommand(
+            packagePath: path,
+            extraArguments: arguments + ["resolve"]
```

Comment:
Will users need to pass arguments that are passed after `resolve`? What do you think if the user is required to pass the full command? That way we don't have to assume whether the arguments that they are passing should go before or after `resolve`?

Reply:
According to `swift package --help`:

```
USAGE: swift package <options> <subcommand>
```

It looks like the options (arguments) should be passed before the subcommand (`resolve` in our case), although I think it will work either way. In my opinion,  we should follow the documentation (`--help`). 

Tuist already passes the package-path option to `swift package`. For example, if we use the additional arguments, we will have something like this:

```
swift package --package-path path/to/package --replace-scm-with-registry resolve
```

I'm not sure if requiring users to provide the full command is what we should be looking for. Perhaps it can be a separate option for more advanced customization needs.

Reply:
The `resolve` command options are tailored for resolving a single package, so I don't think organization will ever need that:
```
OVERVIEW: Resolve package dependencies

USAGE: swift package resolve --version <version> [--branch <branch>] [--revision <revision>] [<package-name>]

ARGUMENTS:
  <package-name>          The name of the package to resolve

OPTIONS:
  --version <version>     The version to resolve at
  --branch <branch>       The branch to resolve at
  --revision <revision>   The revision to resolve at
  --version               Show the version.
  -h, -help, --help       Show help information.
```

I think we should be good with the current approach 👍